### PR TITLE
Store API Call Refactor

### LINKED ISSUE #972: [DevEx]: Refactor Game Store API calls
### Improvement Summary

Refactor Api calls that are all the same

### Detailed Description

Many of the functions inside the Game Store for calling the api are all the same except for the object passed in and the slug of the api address

we can refactor this down to a reusable function like...
```
  handleAPIGet(slug, obj) {
      return new Promise((resolve, reject) => {
        io.socket.get(`/api/game/${slug}`, obj, (res, jwres) => {
          return this.handleGameResponse(jwres, resolve, reject);
        });
      });
    },
    ```

### PR DESCRIPTION
**Pull request by Haviles04**
## Issue number

Relevant [issue number](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- Resolves #972 

## Please check the following

- [x] Do the tests still pass? (see [Run the Tests](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#run-the-tests))
- [x] Is the code formatted properly? (see [Linting (Formatting)](https://github.com/cuttle-cards/cuttle/blob/main/docs/setup-development.md#linting-formatting))
- For New Features:
  - [ ] Have tests been added to cover any new features or fixes?
  - [ ] Has the documentation been updated accordingly?

## Please describe additional details for testing this change
- Created new reusable function
- Refactored old functions to utilize new reusable generic api call function

### CODE REVIEW

**Comment by Haviles04** [comment]
There are a few more functions we could reduce by creating another generic function, but I'm not sure it's worth it for 1 or 2 functions

File: src/stores/game.js [hunk 1]
```diff
@@ -464,107 +470,44 @@ export const useGameStore = defineStore('game', {
...
-          },
-          (res, jwres) => {
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
-      }).then(() => {
+      await this.handleAPIGet('untargetedOneOff', { cardId, opId: this.opponent.id }).then(() => {
         this.waitingForOpponentToCounter = true;
         return Promise.resolve();
       });
```
Comment (itsalaidbacklife) [inline src/stores/game.js:493]: await removes the need for .then()

File: src/stores/game.js [hunk 2]
```diff
@@ -464,107 +470,44 @@ export const useGameStore = defineStore('game', {
...
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
+      await this.handleAPIGet('targetedOneOff', {
+        cardId,
+        targetId,
+        pointId,
+        targetType,
+        opId: this.opponent.id,
       }).then(() => {
```
Comment (itsalaidbacklife) [inline src/stores/game.js:502]: Remove .then()

File: src/stores/game.js [hunk 3]
```diff
@@ -573,230 +516,105 @@ export const useGameStore = defineStore('game', {
...
-          {
-            cardId,
-            opId: this.opponent.id,
-          },
-          (res, jwres) => {
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
-      }).then(() => {
+      await this.handleAPIGet('resolveThree', { cardId, opId: this.opponent.id }).then(() => {
```
Comment (itsalaidbacklife) [inline src/stores/game.js:528]: .then()

File: src/stores/game.js [hunk 4]
```diff
@@ -573,230 +516,105 @@ export const useGameStore = defineStore('game', {
...
-          {
-            cardId: twoId,
-            opId: this.opponent.id,
-          },
-          (res, jwres) => {
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
-      }).then(() => {
+      await this.handleAPIGet('counter', { cardId: twoId, opId: this.opponent.id }).then(() => {
```
Comment (itsalaidbacklife) [inline src/stores/game.js:548]:
```suggestion
      await this.handleAPIGet('counter', { cardId: twoId, opId: this.opponent.id });
```

File: src/stores/game.js [hunk 5]
```diff
@@ -573,230 +516,105 @@ export const useGameStore = defineStore('game', {
...
-            opId: this.opponent.id,
-          },
-          (res, jwres) => {
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
+      await this.handleAPIGet('seven/untargetedOneOff', {
+        cardId,
+        index, // 0 if topCard, 1 if secondCard
+        opId: this.opponent.id,
```
Comment (itsalaidbacklife) [inline src/stores/game.js:587]: .then

File: src/stores/game.js [hunk 6]
```diff
@@ -573,230 +516,105 @@ export const useGameStore = defineStore('game', {
...
-            return this.handleGameResponse(jwres, resolve, reject);
-          },
-        );
+      await this.handleAPIGet('seven/targetedOneOff', {
+        cardId,
+        targetId,
+        pointId,
+        targetType,
+        index, // 0 if topCard, 1 if secondCard
+        opId: this.opponent.id,
```
Comment (itsalaidbacklife) [inline src/stores/game.js:599]: .then()

**Review by itsalaidbacklife — CHANGES_REQUESTED** [review]
Good idea and nice work

File: src/stores/game.js [hunk 7]
```diff
@@ -365,7 +365,24 @@ export const useGameStore = defineStore('game', {
...
+      const url =
+        import.meta.env.VITE_USE_GAMESTATE_API === 'true'
+          ? `/api/gamestate/${this.id}/move/${slug}`
+          : `/api/game/${slug}`;
+      return new Promise((resolve, reject) => {
+        io.socket.request(
+          {
+            method: method,
+            url: url,
+            data: body,
```
Comment (itsalaidbacklife) [inline src/stores/game.js:378]: Let's use the shorthand here. For the same reason let's rename body to data
```suggestion
            method,
            url,
            data,
```

**Review by itsalaidbacklife — APPROVED** [review]
Last nitpick and we're good

**Pull request merged**